apps/backend/api/models/Group.js

```diff
@@ -949,9 +943,38 @@ module.exports = bookshelf.Model.extend(merge({
 
         // Handle post cases - multiple conditions can apply
         if (post) {
+          // Check if it is time to display the stream widget
+          const streamWidget = widgets.find(w => w.get('view') === 'stream')
+          if (streamWidget && !streamWidget.get('order')) {
+            // If there are more than 3 non chat posts, then that stream is flowing
+            const groupPostCount = await Group.postCount(groupId, false)
+            if (groupPostCount > 3) {
+              await ContextWidget.reorder({
+                id: streamWidget.get('id'),
+                parentId: autoAddWidget.get('id'),
+                addToEnd: true,
+                trx
+              })
+            }
+          }
+
+          // Check discussions
+          if (postType === 'discussion') {
+            const discussionsWidget = widgets.find(w => w.get('view') === 'discussions')
+            if (discussionsWidget && !discussionsWidget.get('auto_added')) {
+              await ContextWidget.reorder({
+                id: discussionsWidget.get('id'),
+                parentId: autoAddWidget.get('id'),
+                addToEnd: true,
+                trx
+              })
+            }
+          }
+
           // Check events
           if (postType === 'event') {
             const eventsWidget = widgets.find(w => w.get('view') === 'events')
+            // TODO: instead of checking auto_added shouldnt we just check order? to see if it is added anywhere already?
```

Comment:
If a steward removes one of these sorts of views from the menu, it will have no order. So this check will put it back into the menu, overriding the stewards update. Basically any mutation on a widget will update its auto_added value,  under the assumption that once a steward has moved or changed it, that they have a specific desire for it to be or not be somewhere, and thus we shouldn't reorder it.